wallet/src/de/schildbach/wallet/ui/TransactionResultViewModel.kt

```diff
@@ -30,52 +34,67 @@ import javax.inject.Inject
 
 @FlowPreview
 @HiltViewModel
-class TransactionResultViewModel @Inject constructor(var transactionMetadataProvider: TransactionMetadataProvider) : ViewModel() {
+class TransactionResultViewModel @Inject constructor(
+    private val transactionMetadataProvider: TransactionMetadataProvider,
+    private val walletData: WalletDataProvider,
+    configuration: Configuration
+) : ViewModel() {
 
     companion object {
         val log = LoggerFactory.getLogger(TransactionResultViewModel::class.java)
     }
 
-    private lateinit var transaction: Transaction
+    val dashFormat: MonetaryFormat = configuration.format.noCode()
+
+    val wallet: Wallet?
+        get() = walletData.wallet
+
+    var transaction: Transaction? = null
+        private set
 
     private val _transactionMetadata: MutableStateFlow<TransactionMetadata?> = MutableStateFlow(null)
     val transactionMetadata
         get() = _transactionMetadata.asLiveData()
 
-    fun setTransaction(transaction: Transaction) {
-        this.transaction = transaction
-        monitorTransactionMetadata()
+    fun init(txId: Sha256Hash?) {
```

Comment:
I figure we might as well move the `Transaction` into the `viewModel` completely - less state to keep in the fragment and it's there already anyway.